src/io/contreaddata.c

```diff
@@ -53,14 +56,20 @@ static void __osPackReadData(void) {
 
     __osContPifRam.pifstatus = CONT_CMD_EXE;
     readformat.dummy = CONT_CMD_NOP;
-    readformat.txsize = CONT_CMD_READ_BUTTON_TX;
-    readformat.rxsize = CONT_CMD_READ_BUTTON_RX;
-    readformat.cmd = CONT_CMD_READ_BUTTON;
     readformat.button = 0xFFFF;
     readformat.stick_x = -1;
     readformat.stick_y = -1;
 
     for (i = 0; i < __osMaxControllers; i++) {
+        if ((__osControllerMask & (1 << i)) == 0) {
+            readformat.txsize = CONT_CMD_NOP;
+            readformat.rxsize = CONT_CMD_NOP;
+            readformat.cmd = 0;
```

Comment:
This wastes some space in the PIF packet but if it's not visible in profiling it's not a big deal. What is a big deal is why this works. For a skipped controller, this writes `FF FF FF 00 FF FF FF FF` into the packet, which happens to work since most of the readformat struct is FF which are skip bytes. It might be worth describing what's going on here for future readers.